src/olympia/devhub/tasks.py

```diff
@@ -267,6 +271,52 @@ def handle_file_validation_result(results, file_id, annotate=True):
     return FileValidation.from_json(file_, results)
 
 
+def annotate_new_legacy_addon_restrictions(results):
+    """
+    Annotate validation results to restrict uploads of new legacy
+    (non-webextension) add-ons if specific conditions are met.
+    """
+    metadata = results.get('metadata', {})
+    target_apps = metadata.get('applications', {})
+    target_firefox_version = (
+        target_apps.get('firefox', {}).get('max', None) or
+        target_apps.get('android', {}).get('max', None)
```

Comment:
The [PRD](https://docs.google.com/document/d/1BdiJSYD4aOMClasGxx5wVShHEk8VbxefE3Gf_6jpgkQ/edit#) says *"that support only Firefox, Firefox for Android, or both"*